### PR TITLE
[Fori_loop|While_loop] Placeholder lower torch.while_loop with python dispatch for simple addition test case

### PR DESCRIPTION
**Pull request by ManfeiBai**
This is the step one PR:

Due to `torch.while_loop` is defined in [torch/_higher_order_ops/while_loop.py](https://github.com/pytorch/pytorch/blob/abadbbc4b03138ebe5a892e19ae06eaa41b104d0/torch/_higher_order_ops/while_loop.py#L46), not described in [native_functions.yaml](https://github.com/pytorch/pytorch/blob/main/aten/src/ATen/native/native_functions.yaml), lower `torch.while_loop` via python dispatch like [PR#5763](https://github.com/pytorch/xla/pull/5763)

`TODO`: current status need to patch the `python dispatch` in PyTorch code, we need modify this PR to enable the python dispatch from PyTorch/XLA code

child PR: https://github.com/pytorch/xla/pull/6529

PR resource: https://github.com/pytorch/xla/pull/6527

### CODE REVIEW

**Comment by ManfeiBai** [comment]
CI failed due to `ERROR: Failed to query remote execution capabilities: Unexpected error refreshing access token`, local test passed

**Review by mbzomowski — COMMENTED** [review]
LGTM for the TPU CI portion: https://github.com/pytorch/xla/actions/runs/7980785965/job/21791172250

But I'll leave it to pt/xla folks to approve.

**Review by qihqi — APPROVED** [review]

File: test/test_fori_loop_with_while_loop_simple_add_dispatch_in_torch.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import os
+import unittest
+from typing import Callable, Dict, List
+
+import torch
+import torch_xla
+import torch_xla.experimental.fori_loop
```
Comment (yeounoh) [inline test/test_fori_loop_with_while_loop_simple_add_dispatch_in_torch.py:7]: @ManfeiBai can you leave a brief comment saying that "We need to import the underlying implementation function to register with the dispatcher"

File: torch_xla/experimental/fori_loop.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import torch_xla.utils.utils as xu
+import torch_xla.core.xla_op_registry as xor
+
+from torch._C import DispatchKey
+from torch._ops import HigherOrderOperator
+import torch._higher_order_ops.while_loop
+from torch._higher_order_ops.while_loop import while_loop_op
+
+
+@while_loop_op.py_impl(DispatchKey.XLA)
```
Comment (yeounoh) [inline torch_xla/experimental/fori_loop.py:15]: Great to see that this is working!

File: torch_xla/experimental/fori_loop.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+def while_loop(cond_fn, body_fn, operands):
+  # cond_fn&body_fn: callable
+  # operands: (Tuple of possibly nested dict/list/tuple of tensors)
+  return _xla_while_loop(cond_fn, body_fn, operands)
+
+
+def _xla_while_loop(cond_fn, body_fn, operands):
+
+  def op_fn(internal_x):
+    # TODO(manfei): replace cond_fn_placeholder and body_fn_placeholder after lower fn to xlacomputation and could be in xla::while
```
Comment (yeounoh) [inline torch_xla/experimental/fori_loop.py:25]: nit. "# TODO(manfei) replace cond_fn_placeholder and body_fn_placeholder once xla::while lowering in the backend is available"

**Review by yeounoh — APPROVED** [review]
LGTM, left some minor comments.

File: test/test_fori_loop_with_while_loop_simple_add_dispatch_in_torch.py (hunk 1 above)
Comment (ManfeiBai) [reply test/test_fori_loop_with_while_loop_simple_add_dispatch_in_torch.py:8]: Thanks, updated in the newest commit

File: torch_xla/experimental/fori_loop.py (hunk 3 above)
Comment (ManfeiBai) [reply torch_xla/experimental/fori_loop.py]: Thanks, updated in the newest commit

**Pull request merged**